qutip/__init__.py

```diff
@@ -63,10 +63,25 @@
     os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'
 
 import platform
+import scipy
+from packaging import version as pac_version
 from qutip.utilities import _blas_info
-qutip.settings.eigh_unsafe = (_blas_info() == "OPENBLAS" and
-                              platform.system() == 'Darwin')
-del platform, _blas_info
+
+is_old_scipy = pac_version.parse(scipy.__version__) < pac_version.parse("1.5")
+qutip.settings.eigh_unsafe = (
+                              # macOS OpenBLAS eigh is unstable, see #1288
+                              (_blas_info() == "OPENBLAS" and
+                               platform.system() == 'Darwin')
+                              or
+                              # The combination of
+                              # scipy<1.5 and MKL causes wrong results
+                              # when calling eigh for big matrices.
+                              # See #1495, #1491 and #1498.
+                              (is_old_scipy
+                               and (_blas_info() == 'INTEL MKL'))
+                               )
```

Comment:
Please also fix the formatting of this.  The purpose of the style guidelines is to make code readable - just because something technically passes an automated test doesn't mean it's in a good format.  You can do both at the same time.

Reply:
How do you mean? The `or `  line ? I usually se `SQL` queries written like this. BUt if you want I can change it,

Reply:
Sorry, it's just a minor nitpick, but the spacing - right now it's all crammed into ~35 characters at the far right, when you can space it more legibly as
```python
qutip.settings.eigh_unsafe = (
    # macOS ...
    (_blas_info() == "OPENBLAS" and platform.system() == "Darwin")
    # scipy<1.5 and MKL has bugs...
    or (is_old_scipy and _blas_info() == "INTEL MKL")
)
```

That's all I meant, sorry.